<?xml version="1.0"?>
<launch>
	<arg name="global_frame" default="pioneer/map"/>
	<arg name="tracking_frame" default="pioneer/base_link"/>
	<arg name="object_name" default="animated_box"/>
	<arg name="max_vel_factor" default="1.3"/>
	<arg name="ideal_distance" default="2"/>

	<node name="fake_tracking" type="fake_tracking.py" pkg="tracking_navigation">
	 	<param name="global_frame" value="$(arg global_frame)"/>
		<param name="object_name" value="$(arg object_name)"/>
	</node>
	<node name="relative_pose_publisher" type="relative_pose_publisher.py" pkg="tracking_navigation" output="screen">
		<param name="tracking_frame" value="$(arg tracking_frame)"/>
		<param name="object_name" value="$(arg object_name)"/>
	</node>

	<include file="$(find tracking_navigation)/launch/move_base_pioneer.launch"/>

	<node name="velocity_refactoring" type="velocity_refactoring.py" pkg="tracking_navigation" output="screen">
		<param name="max_vel_factor" value="$(arg max_vel_factor)"/>
		<param name="ideal_distance" value="$(arg ideal_distance)"/>
	</node>
</launch>
